nomad/state/state_store.go

```diff
@@ -2306,6 +2306,21 @@ func (s *StateStore) JobVersionsByID(ws memdb.WatchSet, namespace, id string) ([
 	return s.jobVersionByID(txn, ws, namespace, id)
 }
 
+// JobVersionByTagName returns a Job if it has a Tag with the passed name
+func (s *StateStore) JobVersionByTagName(ws memdb.WatchSet, namespace, id string, tagName string) (*structs.Job, error) {
+	// First get all versions of the job
+	versions, err := s.JobVersionsByID(ws, namespace, id)
+	if err != nil {
+		return nil, err
+	}
+	for _, j := range versions {
+		if j.TaggedVersion != nil && j.TaggedVersion.Name == tagName {
+			return j, nil
+		}
+	}
+	return nil, nil
```

Comment:
my nil pointer sense is tingling! to me `nil, nil` return for `(*pointer, error)` is a capital offense.

shouldn't it be an error if they asked for a tag by name, but no such tag exists?

Reply:
Most of the single-object state store methods return `nil, nil` on non-existing objects, rather than some kind of `NotFound` error.